From c6d9810b22efd6f5c32fa9ae1d7d953a7eee8e92 Mon Sep 17 00:00:00 2001
From: James Turton <james.turton@gmx.com>
Date: Tue, 30 Jun 2026 22:00:01 +0200
Subject: [PATCH] Reduce log rotate threshold

---
 klippy/queuelogger.py | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/klippy/queuelogger.py b/klippy/queuelogger.py
index 413a1934..444d8bbd 100644
--- a/klippy/queuelogger.py
+++ b/klippy/queuelogger.py
@@ -28,16 +28,11 @@ class QueueHandler(logging.Handler):
 
 
 # Class to poll a queue in a background thread and log each message
-class QueueListener(logging.handlers.TimedRotatingFileHandler):
+class QueueListener(logging.handlers.RotatingFileHandler):
     def __init__(self, filename, rotate_log_at_restart):
-        if rotate_log_at_restart:
-            logging.handlers.TimedRotatingFileHandler.__init__(
-                self, filename, when="S", interval=60 * 60 * 24, backupCount=5
-            )
-        else:
-            logging.handlers.TimedRotatingFileHandler.__init__(
-                self, filename, when="midnight", backupCount=5
-            )
+        logging.handlers.RotatingFileHandler.__init__(
+            self, filename, maxBytes=5 * 1024 * 1024, backupCount=2
+        )
         self.bg_queue = queue.Queue()
         self.bg_thread = threading.Thread(target=self._bg_thread)
         self.bg_thread.start()
@@ -64,7 +59,7 @@ class QueueListener(logging.handlers.TimedRotatingFileHandler):
         self.rollover_info.clear()
 
     def doRollover(self):
-        logging.handlers.TimedRotatingFileHandler.doRollover(self)
+        logging.handlers.RotatingFileHandler.doRollover(self)
         lines = [
             self.rollover_info[name] for name in sorted(self.rollover_info)
         ]
