diff --git a/klippy/configfile.py b/klippy/configfile.py
index 691f1eb8..822b483c 100644
--- a/klippy/configfile.py
+++ b/klippy/configfile.py
@@ -791,8 +791,6 @@ class PrinterConfig:
         if get_danger_options().autosave_includes:
             self._save_includes(cfgname, data, set(), gcode)
 
-        # NOW we're safe to check for conflicts
-        self._disallow_include_conflicts(regular_data, cfgname, gcode)
         data = regular_data.rstrip() + autosave_data
         self._write_backup(cfgname, data, gcode)
 
