From 3ae18f60f44efd0dda07eaa2764456c5dc7ac562 Mon Sep 17 00:00:00 2001
From: James Turton <james.turton@gmx.com>
Date: Thu, 7 May 2026 23:40:25 +0200
Subject: [PATCH] Reduce log rotate threshold

---
 klippy/queuelogger.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/klippy/queuelogger.py b/klippy/queuelogger.py
index 413a1934..8d49a572 100644
--- a/klippy/queuelogger.py
+++ b/klippy/queuelogger.py
@@ -36,7 +36,7 @@ class QueueListener(logging.handlers.TimedRotatingFileHandler):
             )
         else:
             logging.handlers.TimedRotatingFileHandler.__init__(
-                self, filename, when="midnight", backupCount=5
+                self, filename, when="H", backupCount=2
             )
         self.bg_queue = queue.Queue()
         self.bg_thread = threading.Thread(target=self._bg_thread)
