diff --git a/src/stm32/Kconfig b/src/stm32/Kconfig
index d45234c..e4a5f1a 100644
--- a/src/stm32/Kconfig
+++ b/src/stm32/Kconfig
@@ -496,6 +496,8 @@ choice
     prompt "Application start offset"
     config STM32_APP_START_20000
         bool "128KiB offset" if MACH_STM32H7 && !MACH_STM32H750
+    config STM32_APP_START_10000
+        bool "64KiB offset" if MACH_STM32F4
     config STM32_APP_START_8000
         bool "32KiB offset" if MACH_STM32F2 || MACH_STM32F4 || MACH_STM32H750
     config STM32_APP_START_4000
@@ -509,6 +511,7 @@ endchoice
 config LAUNCH_APP_ADDRESS
     hex
     default 0x8020000 if STM32_APP_START_20000
+    default 0x8010000 if STM32_APP_START_10000
     default 0x8008000 if STM32_APP_START_8000
     default 0x8004000 if STM32_APP_START_4000
     default 0x8002000 if STM32_APP_START_2000
